
<!DOCTYPE HTML>

<html lang="en">

<head>
  <meta charset="utf-8">
  <title>Std&#58;&#58;bad_exception - C++ - W3cubDocs</title>
  
  <meta name="description" content=" std&#58;&#58;bad_exception is the type of the exception thrown by the C++ runtime in the following situations&#58; ">
  <meta name="keywords" content="std, bad, exception, -, c++, cpp">
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="mobile-web-app-capable" content="yes">
  
  <link rel="canonical" href="http://docs.w3cub.com/cpp/error/bad_exception/">
  <link href="/favicon.png" rel="icon">
  <link type="text/css" rel="stylesheet" href="/assets/application-50364fff564ce3b6327021805f3f00e2957b441cf27f576a7dd4ff63bbc47047.css">
  <script type="text/javascript" src="/assets/application-db64bfd54ceb42be11af7995804cf4902548419ceb79d509b0b7d62c22d98e6f.js"></script>
  <script src="/json/cpp.js"></script>
  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-71174418-1']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body>
	<div class="_app">
	<header class="_header">
  
  <form class="_search">
    <input type="search" class="_search-input" placeholder="Search&hellip;" autocomplete="off" autocapitalize="off" autocorrect="off" spellcheck="false" maxlength="20">
    <a class="_search-clear"></a>
    <div class="_search-tag"></div>
  </form>
  
  <a class="_home-link" href="/" ></a>
  <a class="_menu-link"></a>
  <h1 class="_logo">
    <a href="/" class="_nav-link" title="API Documentation Browser">W3cubDocs</a>
  </h1>
  
  <span class="_logo-sub-nav">/</span><span class="_logo-sub-nav"><a href="/cpp/" class="_nav-link" title="" style="margin-left:0;">C++</a></span>
  
  <nav class="_nav">
    <a href="/app/" class="_nav-link ">App</a>
    <a href="/about/" class="_nav-link ">About</a>
  </nav>
</header>
	<section class="_sidebar">
		<div class="_list">
			
		</div>
	</section>
	<section class="_container ">
		<div class="_content">
			<div class="_page _c">
				
<h1 id="firstHeading" class="firstHeading">std::bad_exception</h1>            <table class="t-dcl-begin"><tbody> <tr class="t-dsc-header"> <th> <div>Defined in header <code><a href="../../header/exception/" title="cpp/header/exception">&lt;exception&gt;</a></code> </div>
</th> <th> </th> <th> </th> </tr> <tr class="t-dcl"> <td class="t-dcl-nopad"> <pre data-language="cpp">class bad_exception;</pre>
</td> <td class="t-dcl-nopad"> </td> <td class="t-dcl-nopad"> </td> </tr>  </tbody></table> <p><code>std::bad_exception</code> is the type of the exception thrown by the C++ runtime in the following situations:</p>
<div class="t-li1">
<span class="t-li">1)</span> If a <a href="../../language/except_spec/" title="cpp/language/except spec">dynamic exception specification</a> is violated and <code><a href="../unexpected/" title="cpp/error/unexpected">std::unexpected</a></code> throws or rethrows an exception that still violates the exception specification, but the exception specification allows <code>std::bad_exception</code>, <code>std::bad_exception</code> is thrown.</div> <div class="t-li1">
<span class="t-li">2)</span> If <code><a href="../exception_ptr/" title="cpp/error/exception ptr">std::exception_ptr</a></code> stores a copy of the caught exception and if the copy constructor of the exception object caught by <code>current_exception</code> throws an exception, the captured exception is an instance of <code>std::bad_exception</code>.</div> <div class="t-inheritance-diagram"> <div class="center"><div class="noresize" style="height: 40px; width: 243px; ">
<map name="ImageMap_1_118147417"><area href="exception" shape="rect" coords="5,4,107,36" alt="cpp/error/exception" title="cpp/error/exception"></map><img alt="std-bad exception-inheritance.svg" src="http://upload.cppreference.com/mwiki/images/0/02/std-bad_exception-inheritance.svg" width="243" height="40" usemap="#ImageMap_1_118147417">
</div></div> <div style="text-align:center;"> <p>Inheritance diagram.</p>
</div> </div>  <h3 id="Member_functions"> Member functions</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div class="t-dsc-member-div t-dsc-member-nobold-div"><div><a href="../bad_exception/bad_exception/" title="cpp/error/bad exception/bad exception"> <span class="t-lines"><span>(constructor)</span></span></a></div></div> </td> <td> constructs the <code>bad_exception</code> object <br> <span class="t-mark">(public member function)</span> </td>
</tr> <tr class="t-dsc"> <td> <div class="t-dsc-member-div"><div><a href="../bad_exception/operator=/" title="cpp/error/bad exception/operator="> <span class="t-lines"><span>operator=</span></span></a></div></div> </td> <td> copies the object <br> <span class="t-mark">(public member function)</span> </td>
</tr> <tr class="t-dsc"> <td> <div class="t-dsc-member-div">
<div><a href="../bad_exception/what/" title="cpp/error/bad exception/what"> <span class="t-lines"><span>what</span></span></a></div>
<div><span class="t-lines"><span><span class="t-cmark">[virtual]</span></span></span></div>
</div> </td> <td> returns the explanatory string <br> <span class="t-mark">(virtual public member function)</span> </td>
</tr> </table> <div class="t-inherited"> <h2 id="Inherited_from_std::exception"> Inherited from <a href="../exception/" title="cpp/error/exception"> std::exception</a> </h2> <h3 id="Member_functions_2"> Member functions</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div class="t-dsc-member-div t-dsc-member-nobold-div">
<div><a href="../exception/~exception/" title="cpp/error/exception/~exception"> <span class="t-lines"><span>(destructor)</span></span></a></div>
<div><span class="t-lines"><span><span class="t-cmark">[virtual]</span></span></span></div>
</div> </td> <td> destructs the exception object <br> <span class="t-mark">(virtual public member function of <code>std::exception</code>)</span>  </td>
</tr> <tr class="t-dsc"> <td> <div class="t-dsc-member-div">
<div><a href="../exception/what/" title="cpp/error/exception/what"> <span class="t-lines"><span>what</span></span></a></div>
<div><span class="t-lines"><span><span class="t-cmark">[virtual]</span></span></span></div>
</div> </td> <td> returns an explanatory string <br> <span class="t-mark">(virtual public member function of <code>std::exception</code>)</span>  </td>
</tr> </table> </div> <h3 id="Example"> Example</h3> <div class="t-example"> <div class="cpp source-cpp"><pre data-language="cpp">#include &lt;iostream&gt;
#include &lt;exception&gt;
#include &lt;stdexcept&gt;
 
void my_unexp() { throw; }
 
void test() throw(std::bad_exception)
{
    throw std::runtime_error("test");
}
 
int main()
{
    std::set_unexpected(my_unexp);
    try {
         test();
    } catch(const std::bad_exception&amp; e)
    {
        std::cerr &lt;&lt; "Caught " &lt;&lt; e.what() &lt;&lt; '\n';
    }
}</pre></div> <p>Output:</p>
<div class="text source-text"><pre data-language="c">Caught std::bad_exception</pre></div> </div>           <div class="_attribution">
  <p class="_attribution-p">
    Â© cppreference.com<br>Licensed under the Creative Commons Attribution-ShareAlike Unported License v3.0.<br>
    <a href="http://en.cppreference.com/w/cpp/error/bad_exception" class="_attribution-link" target="_blank">http://en.cppreference.com/w/cpp/error/bad_exception</a>
  </p>
</div>

			</div>
		</div>
	</section>

	</div>
</body>
</html>
